extension/src/ballistics/bullet/model.rs

```diff
@@ -0,0 +1,216 @@
+                    distribution.sample(&mut self.rng),
+                    distribution.sample(&mut self.rng),
+                    distribution.sample(&mut self.rng),
+                );
+                let coef = 1.0 - self.transonic_stability_coefficient;
+                let true_speed = true_velocity.magnitude();
+                true_velocity += offset * coef;
+                true_velocity = true_velocity.normalize() * true_speed;
+            }
+
+            let drag = if self.ballistic_coefficients.len() == self.velocity_boundaries.len() + 1 {
+                let mut ballistic_coefficient = self.ballistic_coefficients[0];
+                for (i, boundry) in self.velocity_boundaries.iter().enumerate() {
+                    if true_velocity.magnitude() < *boundry {
+                        ballistic_coefficient = self.ballistic_coefficients[i + 1];
+                        break;
+                    }
+                }
+
+                ballistic_coefficient = calculate_atmospheric_correction(
+                    ballistic_coefficient,
+                    temperature,
+                    pressure,
+                    self.humidity,
+                    self.atmosphere_model,
+                );
+
+                calculate_retard(
+                    self.drag_function,
+                    ballistic_coefficient,
+                    true_velocity.magnitude(),
+                    speed_of_sound(temperature),
+                )
+            } else {
+                let air_density = calculate_air_density(temperature, pressure, self.humidity);
+                let air_friction = self.air_friction * air_density / STD_AIR_DENSITY_ICAO;
+
+                -air_friction * true_velocity.magnitude_squared()
+            };
+
+            let accel = true_velocity.normalize() * drag;
+
+            velocity_offset -= accel * dt;
+            bullet_velocity -= accel * dt;
+
+            if tof > 0.0 {
+                let bullet_dir = bullet_velocity.x().atan2(bullet_velocity.y());
+                let drift_accel = self.twist_direction
+                    * (0.048_225_1 * (self.stability_factor + 1.2))
+                    / tof.powf(0.17);
+                let drift_velocity = 0.058_102_5 * (self.stability_factor + 1.2) * tof.powf(0.83);
+                let drag_correction = (drift_velocity / true_velocity.magnitude()) * drag;
+                let magnitude = (drift_accel + drag_correction) * dt;
+                let offset = Vector3::new(
+                    (bullet_dir + PI / 2.0).sin() * magnitude,
+                    (bullet_dir + PI / 2.0).cos() * magnitude,
+                    0.0,
+                );
+                velocity_offset += offset;
+                bullet_velocity += offset;
+            }
+
+            let lat = self.latitude;
+            let accel = Vector3::new(
+                2.0 * EARTH_ANGULAR_SPEED
+                    * (bullet_velocity.y() * lat.sin() - bullet_velocity.x() * lat.cos()),
```

Comment:
[c++ src](https://github.com/acemod/ACE3/blob/master/extensions/advanced_ballistics/AdvancedBallistics.cpp#L611)
```suggestion
                    * (bullet_velocity.y() * lat.sin() - bullet_velocity.z() * lat.cos()),
```